File: BackEnd/src/domain/interfaces/usecases/community/ICreatePostUseCase.ts
/**
 * @file ICreatePostUseCase.ts
 * @description Interfaz del caso de uso para crear un post en la comunidad.
 * @module Community
 * @layer Domain
 */

import type { CreatePostRequestDto } from '../../../dtos/community/create-post-request.dto.js';
import type { PostResponseDTO } from '../../../dtos/community/post-response.dto.js';

/**
 * Contrato del use case CreatePost.
 */
export interface ICreatePostUseCase {
  /**
   * Ejecuta el caso de uso.
   *
   * @param authorId — Id del usuario que crea el post.
   * @param data — Datos del post a crear.
   * @returns PostResponseDTO del post creado.
   */
  execute(authorId: string, data: CreatePostRequestDto): Promise<PostResponseDTO>;
}

File: FrontEnd/lib/domain/interfaces/usecases/community/i_create_post_use_case.dart
/// @file i_create_post_use_case.dart
/// @description Interfaz: Crea un nuevo post en la comunidad.
/// @module Community
/// @layer Domain
library;

import '../../../dtos/community/create_post_request_dto.dart';
import '../../../entities/post.dart';

abstract interface class ICreatePostUseCase {
  /// Crea un nuevo post en la comunidad.
  ///
  /// [param] dto — DTO con el contenido, imageUrl opcional y plantId opcional.
  /// [returns] [Post] recién creado con datos del servidor.
  Future<Post> execute(CreatePostRequestDto dto);
}
